### PR TITLE
Improve Kebab-casing performance

### PR DESCRIPTION
**Pull request by lahma**
* renamed ToPascalCase, accepting the delimiter to start upper-casing after
* don't build string builder if not going to change the string
* less LINQ

Only problematic thing is exposing new public API and removing old one.

### CODE REVIEW

**Comment by sebastienros** [comment]
There is an issue on the functional tests. I will try your branch locally.

**Comment by lahma** [comment]
Thanks, feel free to tweak as needed if you have the time. It's getting bed-time around here so it'll take some time to respond to feedback. I do love a good kebab/döner, shame to remove the name 😉

**Comment by sebastienros** [comment]
Found the issue, the dictionary you are now creating is `<string,string>` which triggered a different overload in `Arguments.From`. I fixed `Arguments` to handle this kind of dic too. Restarted the CI to check it works now.

File: src/OrchardCore/OrchardCore.DisplayManagement/TagHelpers/BaseShapeTagHelper.cs [hunk 1]
```diff
@@ -36,10 +42,14 @@ protected BaseShapeTagHelper(IShapeFactory shapeFactory, IDisplayHelper displayH
         public override async Task ProcessAsync(TagHelperContext tagHelperContext, TagHelperOutput output)
         {
             // Extract all attributes from the tag helper to
-            var properties = output.Attributes
-                .Where(x => !InternalProperties.Contains(x.Name))
-                .ToDictionary(x => LowerKebabToPascalCase(x.Name), x => (object)x.Value.ToString())
-                ;
+            var properties = new Dictionary<string, string>();
+            foreach (var pair in output.Attributes)
```
Comment (jtkech) [inline src/OrchardCore/OrchardCore.DisplayManagement/TagHelpers/BaseShapeTagHelper.cs]: i think it needs to be the following to target the right `Arguments.From(properties)` method

    var properties = new Dictionary<string, object>();

**Comment by jtkech** [comment]
Found it too late

**Pull request merged**